archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -789,21 +818,58 @@ public JavaMethod getMethod(String name, String... parameters) {
         return allMethods.get();
     }
 
+    /**
+     * @return The constructor with zero parameters (the default constructor).
```

Comment:
I'd drop `(the default constructor)` here to avoid confusion with the implicit constructor from [JLS §8.8.9](https://docs.oracle.com/javase/specs/jls/se14/html/jls-8.html#jls-8.8.9) that is automatically added in absence of explicit constructor declarations – which, for inner classes, actually has a parameter.

Reply:
Same for `tryGetConstructor()`.

Reply:
Ah, you're right, I forgot the inner classes. In this case the "default constructor"-note would be confusing.